service/worker/scheduler/workflow.go

```diff
@@ -836,6 +840,14 @@ func (s *scheduler) processWatcherResult(id string, f workflow.Future, long bool
 		s.logger.Warn("just-closed workflow not found in running list", "workflow", id, "long", long)
 	}
 
+	// update workflow execution status in RecentActions
+	if s.hasMinVersion(ActionResultIncludesStatus) {
```

Comment:
Hmm, I was imagining always adding it here and then doing a version check in getListInfo to _remove_ the field, which is the only place we have to be careful about being compatible on replay.

If we do it here... it's clearly upgrade-compatible, since we don't set the new field until we're using the new logic version. Is it downgrade-compatible? I think it is, though I'm finding it a little harder to reason about.

In any case, can we add a replay test history for this to be sure? (i.e. generate a history with the current version set to this new version and make sure it replays on itself and the current version too)

Reply:
Added, verified the new replay test also works on `main` without my changes.